azure-pipelines.yml

```diff
@@ -7,10 +7,56 @@ pool:
   vmImage: 'Ubuntu 16.04'
 
 steps:
-- script: echo Hello, world!
-  displayName: 'Run a one-line script'
+- task: UsePythonVersion@0
+  inputs:
+    versionSpec: 3.6
+    architecture: 'x64'
+
+- task: PythonScript@0
+  displayName: 'Template Runner'
+  inputs:
+    scriptSource: 'inline'
+    script: |
+      """Search all subdirectories for `Runner.py`."""
+      from glob import iglob
+      from os import path
+      import os
+      # Python >= 3.6
+
+      os.environ["PS_BATCH_ACCOUNT_KEY"] = "$(PS_BATCH_ACCOUNT_KEY)" 
+      os.environ["PS_BATCH_ACCOUNT_NAME"] = "$(PS_BATCH_ACCOUNT_NAME)"
+      os.environ["PS_BATCH_ACCOUNT_SUB"] = "$(PS_BATCH_ACCOUNT_SUB)"
+      os.environ["PS_BATCH_ACCOUNT_URL"] = "$(PS_BATCH_ACCOUNT_URL)"
+      os.environ["PS_SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID"] = "$(PS_SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID)" 
+      os.environ["PS_SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE"] = "$(PS_SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE)" 
+      os.environ["PS_SERVICE_PRINCIPAL_CREDENTIALS_TENANT"] = "$(PS_SERVICE_PRINCIPAL_CREDENTIALS_TENANT)"
+      os.environ["PS_SERVICE_PRINCIPAL_CREDENTIALS_SECRET"] = "$(PS_SERVICE_PRINCIPAL_CREDENTIALS_SECRET)"
+      os.environ["PS_STORAGE_ACCOUNT_KEY"] = "$(PS_STORAGE_ACCOUNT_KEY)"
+      os.environ["PS_STORAGE_ACCOUNT_NAME"] = "$(PS_STORAGE_ACCOUNT_NAME)"
+      
+      print(os.listdir())
+      os.listdir()
+ 
+      Runner_py = next(iglob(path.join('**', 'Runner.py'), recursive=True), None)
+      if not Runner_py:
+          raise SystemExit('Could not find a Django project')
```

Comment:
? what is this doing?

Reply:
basically saying if we can't find the runner.py file throw an error.

Reply:
💡 I think what tim is coming at is the error message says a word called : `django` Django is a different py `framework` read here: ``https://www.djangoproject.com/

Reply:
wondering why it couldn't find this file here though?

Reply:
I have a feeling that **use** of `async` and `exception handling` is vastly used in these files as loose concepts? @matthew-rollitt Please add comments and Details at places where things are not clear. 

2 things will benefit:

* We will be able to pin point or recommend if you are not sure.
* You will be clear how the runner is working end-to-end.